diff --git a/src/nvfs-core.c b/src/nvfs-core.c
index eb447ff..165b473 100644
--- a/src/nvfs-core.c
+++ b/src/nvfs-core.c
@@ -1967,7 +1967,8 @@ done:
 static inline bool nvfs_need_fallocate(struct inode *inode) {
 	unsigned long magic = inode->i_sb->s_magic;
 	return ((magic != NFS_SUPER_MAGIC) &&  (magic != SCATEFS_SUPER_MAGIC) &&
-		(magic != LUSTRE_SUPER_MAGIC) && (magic != BEEGFS_SUPER_MAGIC));
+		(magic != LUSTRE_SUPER_MAGIC) && (magic != BEEGFS_SUPER_MAGIC) &&
+		(magic != RAILFS_SUPER_MAGIC));
 }
 
 long nvfs_io_start_op(nvfs_io_t* nvfsio)
diff --git a/src/nvfs-core.h b/src/nvfs-core.h
index 309ed7c..2f61f76 100644
--- a/src/nvfs-core.h
+++ b/src/nvfs-core.h
@@ -250,6 +250,7 @@ void nvfs_io_process_exiting(nvfs_mgroup_ptr_t nvfs_mgroup);
 #define LUSTRE_SUPER_MAGIC 0x0bd00bd0U
 #define BEEGFS_SUPER_MAGIC  0x19830326U
 #define SCATEFS_SUPER_MAGIC 0x53544653
+#define RAILFS_SUPER_MAGIC  0x5241494cU
 #define NVFS_MAY_SLEEP()                  (!irqs_disabled() && !in_interrupt() && !in_atomic() && !in_nmi())
 
 static inline unsigned int get_major(struct inode *inode) {
diff --git a/src/nvfs-dma.c b/src/nvfs-dma.c
index 1e74785..a7dce23 100644
--- a/src/nvfs-dma.c
+++ b/src/nvfs-dma.c
@@ -200,6 +200,18 @@ struct module_entry modules_list[] = {
 		&nvfs_dev_dma_rw_ops
 	},
 
+	{
+		1,
+		0,
+		NVFS_PROC_MOD_RAILFS_KEY,
+		0,
+		"railfs_register_nvfs_dma_ops",
+		0,
+		"railfs_unregister_nvfs_dma_ops",
+		0,
+		&nvfs_dev_dma_rw_ops
+	},
+
 	{
 		0,
 		0,
diff --git a/src/nvfs-dma.h b/src/nvfs-dma.h
index 4ad63e0..c1918af 100644
--- a/src/nvfs-dma.h
+++ b/src/nvfs-dma.h
@@ -69,6 +69,7 @@
 #define NVFS_PROC_MOD_NFS_KEY           "rpcrdma"
 #define NVFS_PROC_MOD_WEKAFS_KEY        "wekafsio"
 #define NVFS_PROC_MOD_SCATEFS_KEY       "scatefs"
+#define NVFS_PROC_MOD_RAILFS_KEY        "railfs"
 
 
 struct nvfs_dma_rw_ops {
